compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1856,7 +1863,56 @@ impl<'a> Resolver<'a> {
                         )
                     },
                 )
-            };
+            });
+
+            // 3. check whether the name refers to an item in local scope
+            // then delay to check whether methond name is an associated function or not
+            // For example:
+            // ```
+            // struct Foo;
+            // impl Foo {
+            //    fn bar(&self) {}
+            // }
+            //
+            // fn main() {
+            //    let foo = Foo {};
+            //    foo::bar(); // suggest to foo.bar();
+            //}
+            //```
+            if suggestion.is_none() &&
+                let Some(ribs) = ribs &&
+                let Some(LexicalScopeBinding::Res(Res::Local(local_id))) = self.resolve_ident_in_lexical_scope(
+                    ident,
+                    ValueNS,
+                    parent_scope,
+                    None,
+                    &ribs[ValueNS],
+                    ignore_binding,
+                )
+                {
+                    let sm = self.session.source_map();
+                    // only check path with two segments, like `foo::bar(..)`
+                    if let Ok(span) = sm.span_extend_while(ident.span.shrink_to_hi(), |c| c != ' ') &&
+                        let Ok(snippet) = sm.span_to_snippet(span) &&
+                            snippet.starts_with("::") &&  snippet.matches("::").count() == 1 {
+                        let local_span = *self.pat_span_map.get(&local_id).unwrap();
+                        let mut err = self.session.struct_span_err(
```

Comment:
Why are you making a new error here? Is it possible to pass this down to where the original error (`use of undeclared crate or module`) is being created?

Reply:
This is because I need to stash it, and if the method checking failed, I want cancel it, but the previous one `use of undeclared crate or module` will always emit?

Reply:
oops, seems need many more changes, since `report_path_resolution_error` only report the error message and suggestions.

Reply:
maybe this function is not the proper position to add this check...😛

Reply:
Maybe so... I would prefer if we did it the correct way. Is it possible to investigate?

Reply:
Yes, let me try it.

Reply:
Code updated! 
There are some test case result changed, since the diagnostic order change.